src/language-html/ast.js

```diff
@@ -12,25 +12,31 @@ const NODES_KEYS = {
 // https://github.com/microsoft/TypeScript/issues/26811
 
 class Node {
-  constructor(props = {}) {
+  constructor(props = {}, parent) {
```

Comment:
This `parent` argument is weird to me, especially when use `node.clone(parent)`, maybe `new Node({..., parent})`?